client/extensions/tga-sign-off/src/utils.ts

```diff
@@ -45,9 +45,15 @@ export function hasUserSignedOff(item: IArticle): boolean {
     return false;
 }
 
+function getAuthorsWithEditorRole(authors: Array<IAuthor>): Array<IAuthor> {
+    return authors.filter(author => author.role === 'editor');
+}
+
 export function getListAuthorIds(item: IArticle): Array<IUser['_id']> {
+    const authorsWithEditorRole = getAuthorsWithEditorRole(item.authors ?? []);
+
     // @ts-ignore
-    return (item.authors ?? [])
+    return authorsWithEditorRole
```

Comment:
all good, but I'd not add a separate function:

```
export function getListAuthorIds(item: IArticle): Array<IUser['_id']> {
    const {notNullOrUndefined} = superdesk.helpers;

    return (item.authors ?? [])
        .filter(author => author.role === 'editor')
        .map((author) => author.parent)
        .filter(notNullOrUndefined);
}
```

I've also changed the function a bit so `@ts-ignore` is not needed. @MarkLark86 check it out.

Reply:
Why filter twice? 

```ts
export function getListAuthorIds(item: IArticle): Array<IUser['_id']> {
    const {notNullOrUndefined} = superdesk.helpers;

    return (item.authors ?? [])
        .map((author) => author.parent)
        .filter((author) => notNullOrUndefined(author) && author.role === 'editor');
}
```

Reply:
your version wouldn't work because `author` as filter argument is a string and you later try to access `author.role`